subsys/net/lib/nrf_cloud/Kconfig.nrf_cloud_agps

```diff
@@ -3,8 +3,28 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 
-config NRF_CLOUD_AGPS
+menuconfig NRF_CLOUD_AGPS
 	bool "nRF Cloud Assisted GPS (A-GPS)"
 	depends on MODEM_INFO
 	depends on MODEM_INFO_ADD_NETWORK
 	select CJSON_LIB
+
+if NRF_CLOUD_AGPS
+
+config NRF_CLOUD_AGPS_FILTERED
+	bool "Request only visible satellite ephemerides"
+	help
+	  Normally A-GPS requests ephemerides for all GPS constellation
+	  satellites. Setting this option reduces that set to only
+	  satellites at or above the threshold elevation angle.
```

Comment:
How does the cloud handle this - does it send ephemerides for the satellites above the threshold at the time of request or does it send ephemerides for all the satellites that will rise above the threshold during the validity period of the ephemeris to be provided?

Reply:
Good Question.  Cole has asked RX Network for details.  We'll let you know.

Reply:
Heard back today:

```
The filtering functionality is implemented to return the satellites above the elevation mask at the time the server processes the request.

It’s designed to ensure that the satellites received are able to be all usable for positioning at that point in time.
```

I have a question pending about how long the ephems are valid for, but haven't heard back yet. I suspect it's not fixed and based on how far over the satellites are already.

Reply:
Thanks. If our fix interval is long enough to trigger an A-GPS data request every time, this does not make any different - we have all the info we need for now. But if we attempt to get another fix using the previously downloaded filtered ephemeris it is likely that new satellites have appeared on the sky for which we don't have ephemeris and old satellites have disappeared from view. This may lead to a PVT solution with less satellites or even no fix at all. Another issue is that if GNSS is missing ephemeris for more than two or so SVs, it keeps requesting for A-GPS data. Filtering A-GPS requests pretty much guarantees that the modem will request assistance every time it is allowed to do so, i.e., every 60 minutes. Taking all this into account, I think we should state very clearly in the documentation that filtering A-GPS requests should only be used when the fix interval is longer than the ephemeris validity time, i.e., we need to do A-GPS request every time we make a fix anyway. Any savings in energy or data consumption due to filtering ephemeris will be lost many times over if we need to make a second request to download something we could have downloaded the first time but we filtered it out. This is mainly related to relatively large energy overhead when setting up the LTE connection.